polytracker/include/taintdag/control_flow_log.h

```diff
@@ -0,0 +1,88 @@
+
+/*
+ * Copyright (c) 2022-present, Trail of Bits, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed in accordance with the terms specified in
+ * the LICENSE file found in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include "taintdag/outputfile.h"
+#include "taintdag/section.h"
+#include "taintdag/taint.h"
+#include "taintdag/util.h"
+
+namespace taintdag {
+
+namespace details {
+// A uint32_t varint encoded by setting highest bit for all but the final byte.
+// Requires up to 5 bytes of storage as each output byte uses 7 input bits.
+// Total maximum need is floor(32/7) = 5. Returns number of bytes required.
+size_t varint_encode(uint32_t val, uint8_t *buffer) {
```

Comment:
Can you explain this encoding scheme to me?

Reply:
OK, I found an explanation [here](https://carlmastrangelo.com/blog/lets-make-a-varint). I'm asking myself if this is worth the extra code here. Maybe we could add generic standard compression functionality and compress whole sections instead?